cmd/logger/message/log/entry.go

```diff
@@ -18,8 +18,8 @@ package log
 
 // Args - defines the arguments for the API.
 type Args struct {
-	Bucket   string            `json:"bucket,omitempty"`
-	Object   string            `json:"object,omitempty"`
+	Bucket   string            `json:"bucket,omitempty" shadow:"BUCKET"`
```

Comment:
Why do these kinds of annotations? - It is perhaps better to write a custom UnmarshalJSON which simply avoids filling in the Bucket, Object value based on some settings.  These can be simplified I believe. It is harder to maintain a json annotation code parser.

Reply:
> Why do these kinds of annotations? - It is perhaps better to write a custom UnmarshalJSON which simply avoids filling in the Bucket, Object value based on some settings. These can be simplified I believe. It is harder to maintain a json annotation code parser.

I am not sure how UnmarshalJSON would fit here, would you elaborate more ?

Basically I wanted the most generic solution to avoid human errors here, but I agree it is kind of complicated to read code using reflect package.

Reply:
Adding new annotations is harder and cryptic. It is very simple to implement UnmarshalJSON to behave differently because this is simply just a struct definition ... but you can write Unmarshallers in cmd/logger/ which already has a global awareness if anonymousFlag is set or not.